vipaccess/provision.py

```diff
@@ -164,30 +170,52 @@ def decrypt_key(token_iv, token_cipher):
 
     return otp_key
 
+def decode_secret_hex(secret):
+    '''Get Secret in Hex For Yubikey'''
+    return binascii.b2a_hex(secret).decode('utf-8')
+
+
 def generate_otp_uri(token, secret):
     '''Generate the OTP URI.'''
     token_parameters = {}
-    token_parameters['otp_type'] = urllib.quote('totp')
     token_parameters['app_name'] = urllib.quote('VIP Access')
     token_parameters['account_name'] = urllib.quote(token['id'])
-    token_parameters['parameters'] = urllib.urlencode(
-        dict(
-            secret=base64.b32encode(secret).upper(),
-            digits=token['digits'],
-            period=token['period'],
-            algorithm=token['algorithm'],
-            issuer='Symantec'
+    if urllib.quote(token['id']).startswith('VSMB'):
+        token_parameters['otp_type'] = urllib.quote('hotp')
+        token_parameters['parameters'] = urllib.urlencode(
+            dict(
+                secret=base64.b32encode(secret).upper(),
+                digits=token['digits'],
+                count='2',
+                algorithm=token['algorithm'],
+                issuer='Symantec'
+                )
+            )
+    else:
+        token_parameters['otp_type'] = urllib.quote('totp')
+        token_parameters['parameters'] = urllib.urlencode(
+            dict(
+                secret=base64.b32encode(secret).upper(),
+                digits=token['digits'],
+                period=token['period'],
+                algorithm=token['algorithm'],
+                issuer='Symantec'
+                )
             )
-        )
 
     return 'otpauth://%(otp_type)s/%(app_name)s:%(account_name)s?%(parameters)s' % token_parameters
 
 def check_token(token_id, secret, session=requests):
     '''Check the validity of the generated token.'''
-    otp = totp(binascii.b2a_hex(secret).decode('utf-8'))
     test_url = 'https://vip.symantec.com/otpCheck'
```

Comment:
That was my mistake to put the test URL in here twice. I meant to use the "file scope" one from [line 46](https://github.com/dlenski/python-vipaccess/pull/10/files#diff-36c792869521328d851e6ba1a9807b82R46).